### PR TITLE
Building Compassionate Software

### LINKED ISSUE #248: Psychological Safety in Software Teams
The idea s simple: teams made of people who feel safe to be themselves and to ask questions/point out mistakes perform better than teams without these traits. 

Resources:

- [NYT article on Project Aristotle](http://www.nytimes.com/2016/02/28/magazine/what-google-learned-from-its-quest-to-build-the-perfect-team.html)
- [Quartz Article on Psychological Safety](http://qz.com/625870/after-years-of-intensive-analysis-google-discovers-the-key-to-good-teamwork-is-being-nice/)
- [Amy Edmunson's talk on her research](https://www.youtube.com/watch?v=LhoLuui9gX8)
- [OSFeels talk](https://www.youtube.com/watch?v=OdHH5KWGTNo)
- [Quartz article on empathy helping students / lack of empathy hurting students](http://qz.com/622749/teens-do-better-in-science-when-they-know-einstein-and-curie-also-struggled/)

What I need now is a compelling story around psychological safety to tell.

### PR DESCRIPTION
**Pull request by ashfurrow**
This is only a first draft, this post is important enough to let it percolate and get feedback on.

Fixes #248.

### CODE REVIEW

File: source/blog/2016-08-20-building-compassionate-software.html.markdown [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+1. **Feelings Matter.** Before we talk about feelings, we should discuss why exactly they matter. There's a lot of evidence here that I'm excited to talk about.
+2. **Teams with Psychological Safety Perform Better.** I'm going to describe what psychological safety means and what it looks like.
+3. **How to Implement Psychological Safety on Your Team.** After we have a firm grasp on feelings and psychological safety, I want to discuss some ways to start improving your team's performance and dynamics.
+
+I've had this topic on my mind for a long time, and the 
+
+## Feelings Matter
+
+So I've [written about this before](https://ashfurrow.com/blog/empathetic-civilization/), but I'll say it again: _feelings matter_. It might sound obvious to you, but it really does need to be repeated. Compassionate software can't be built without compassion for each other.
```
Comment (ashfurrow) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: Does it really sound obvious?

File: source/blog/2016-08-20-building-compassionate-software.html.markdown [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+(READMORE)
+
+I'm going to take us through three main points:
+
+1. **Feelings Matter.** Before we talk about feelings, we should discuss why exactly they matter. There's a lot of evidence here that I'm excited to talk about.
+2. **Teams with Psychological Safety Perform Better.** I'm going to describe what psychological safety means and what it looks like.
+3. **How to Implement Psychological Safety on Your Team.** After we have a firm grasp on feelings and psychological safety, I want to discuss some ways to start improving your team's performance and dynamics.
+
+I've had this topic on my mind for a long time, and the
```
Comment (orta) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: not finished

File: source/blog/2016-08-20-building-compassionate-software.html.markdown (hunk 1 above)
Comment (orta) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: FWIW, you don't define compassionate software

File: source/blog/2016-08-20-building-compassionate-software.html.markdown [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+- Staying non-judgemental.
+- Communicating that you understand.
+
+These are things we can do, things we can _choose_ to do. We can practice, we can set a Google Calendar reminder to reflect on recent meetings with the intent to empathize with our peers.
+
+This is how someone empathizes, which is core to feelings, which matter. Feelings are important. Now I want to explore what it means to be a member of a team where feelings are prioritized.
+
+## Teams with Psychological Safety Perform Better
+
+So you want to be a 10x developer, eh? You may have heard that the 10x developer is a myth, but that's not true: a 10x developer is someone who makes the five developers around them each twice as productive. You can be a 10x developer by making sure that your team has psychological safety.
```
Comment (orta) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: this point is great

File: source/blog/2016-08-20-building-compassionate-software.html.markdown [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  1. Conversational turn-taking.
+  2. Average emotional sensitivity.
+
+Conversational turn-taking is a measure of how often people in a conversation switch from talking to listening. One member of a team who dominates the conversation is risking the psychological safety of the entire team. Everyone needs to feel safe having their say, and to revisit a conversation later necessary.
+
+Average emotional sensitivity is a bit trickier. Emotional sensitivity is basically a measure of how empathetic some is to another's feelings. For example, how often does one colleague notice another colleague is having a difficult time? And when they notice, do they try to understand? Do they stay non-judgemental? And do they communicate that understanding?
+
+Psychological safety really ought to be expected at your workplace. At any workplace, really, for two reasons:
+
+1. It makes team members feel safer, like everyone is welcome.
```
Comment (orta) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: feels a bit valley-girl, perhaps "making everyone feel welcome" ?

File: source/blog/2016-08-20-building-compassionate-software.html.markdown [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+Okay. So feelings matter, and psychological safety is why high-performing teams do so well. But how would one go about improving their team's dynamic? Where do we start?
+
+I'm glad you asked.
+
+## How to Implement Psychological Safety on Your Team
+
+Psychological safety is awesome! How do we "do psychological safety" though? That's an interesting question. First we need to talk about the two scenarios you're likely to find yourself in.
+
+If you're a team leader, there's a lot you can do to improve you team's dynamic and – as a consequence – you're team's performance. Let's take a look at those steps after we discuss how individual contributors can help their teams, too.
```
Comment (orta) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: `you're` -> `your`

File: source/blog/2016-08-20-building-compassionate-software.html.markdown (hunk 2 above)
Comment (ashfurrow) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: Nice, thanks.

File: source/blog/2016-08-20-building-compassionate-software.html.markdown [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+1. **Admit fallibility** and normalize struggle.
+2. Frame all work as **learning experiences**.
+3. **Model curiosity** by creating a necessity for voice.
+
+First, remember that everyone struggles and everyone makes mistakes. If you, as a team lead, make this the norm, then that sends a message to team members that it's okay to make mistakes. Honestly, it's pretty straightforward: you want your team to feel safe when things go wrong, so make sure to act normal when you make a mistake.
+
+Next, all work your team performs should be primarily modelled as exercises in learning. Because that's what they are; when a team build something, you're all really just _learning how to build something_ as a team. The byproduct of this is the thing that happened to built. 
+
+The product a team builds is important to the business' success, so it may seem counterintuitive to place a higher priority on the learning experience of a team than on building the product itself. But remember: by doing this, you're helping to increase the performance of your team so in turn, they're able to build a better product, faster and with fewer bugs. The evidence shows it makes business sense.
```
Comment (orta) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: > But remember: by doing this, you're helping to increase the performance of your team so in turn

The "so in turn" feels like a separate thing. Maybe a `-`?

File: source/blog/2016-08-20-building-compassionate-software.html.markdown (hunk 3 above)
Comment (ashfurrow) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: I couldn't find a source for it, do you know who said it first? @dbgrandi?

File: source/blog/2016-08-20-building-compassionate-software.html.markdown [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+This advice is really built upon empathy, which means there are a few other common sense tidbits that accompany it:
+
+- Watch out for people getting interrupted in meetings. When you see it, say "hang on, I want to hear the rest of what they have to say."
+- Don't pressure people into providing immediate feedback. Instead of asking on the spot, give time for reflective feedback. "I'll type up what we've discussed and send it to everyone, let me know what isn't clear." 
+- Allow your team to revisit discussion if someone feels their voice wasn't heard.
+
+Psychological safety can be a differentiator at your workplace. It's hard to retain good developers, and it's harder to find them in the first place. Working in a safe environment, where everyone feels like they can ask questions and where everyone is able to do their best work, well that sounds awesome, doesn't it? Implement these suggestions so that your workplace stands out to prospective developers. 
+
+This can be a workplace differentiator, likely more attractive than free snacks or a foosball table to prospective and current colleagues.
```
Comment (orta) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: one thing about this point is that it's hard to be able to show that to potential hires, can you think of ways in which that could be demoable?

File: source/blog/2016-08-20-building-compassionate-software.html.markdown [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+YOUTUBE43 LhoLuui9gX8
+
+Developers need to be able to ask questions when they don't understand something. We need to feel free to suggest ideas or concerns, to be able to point out and admit mistakes. This is really necessary for development teams, and _especially_ necessary in resource-strained startups where missteps could cost the company.
+
+There was this one time at Artsy where I was behind schedule on a big feature, which was delaying me from starting work on [something really important](http://artsy.github.io/blog/2016/08/09/the-tech-behind-live-auction-integration/). I sat down with my team, and we had _just_ started reviewing the designs for all the stuff I had to finish when a designer asked "what if... we just... didn't do any of it?" We hadn't really considered whether the feature I was behind on was worth delaying the next project. A designer from outside our team was comfortable challenging our assumptions. They were right, we dropped the delayed feature and moved directly on to what was more important.
```
Comment (orta) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: "what if... we just... didn't do any of it?" tense is off, they would have said "don't"

File: source/blog/2016-08-20-building-compassionate-software.html.markdown (hunk 3 above)
Comment (orta) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: I don't know it I'm afriad

Comment (dbgrandi) [inline source/blog/2016-08-20-building-compassionate-software.html.markdown]: Not sure who said it first, but I saw it in a tweet.

https://twitter.com/peterseibel/status/512615519934230528

**Pull request merged**